week4/Program.cs

```diff
@@ -0,0 +1,140 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+// A POST endpoint that calls POST users/add with a record with FirstName, LastName and Age
+app.MapPost("users/new", async (User addUser) => await CreateUser(addUser));
+
+async Task<User?> CreateUser(User addUser)
+{
+    try
+    {
+        var response = await new HttpClient().PostAsJsonAsync("https://dummyjson.com/users/add", addUser);
+        if (response == null)
+        {
+            return (User)Results.NotFound();
+        }
+        var newUser = await response.Content.ReadFromJsonAsync<User>();
+        return newUser;
+    }
+    catch (Exception e)
+    {
+        throw new Exception(e.Message);
+    }
+};
+
+// A POST endpoint that that calls Post products/add with a record with Title and Price
+app.MapPost("products/new", async (Product addProduct) => await CreateProduct(addProduct));
+
+async Task<Product?> CreateProduct(Product addProduct)
+{
+    try
+    {
+        var response = await new HttpClient().PostAsJsonAsync("https://dummyjson.com/products/add", addProduct);
+        var newProduct = await response.Content.ReadFromJsonAsync<Product>();
+        return newProduct;
+    }
+    catch (Exception e)
+    {
+        throw new Exception(e.Message);
+    }
+};
+
+// A POST endpoint that takes a lists of ids and retrieves all of the users with those ids from the GET users (Id, FirstName, LastName and Age)
+app.MapPost("/userIds", async (List<int> userIds) => await GetUsersAsync(userIds));
+
+async Task<List<User>> GetUsersAsync(List<int> userIds)
+{
+    var getUserTasks = new List<Task<User>>();
+    foreach (int userId in userIds)
+    {
+        getUserTasks.Add(GetUserAsync(userId));
+    }
+    var users = await Task.WhenAll(getUserTasks);
+    return new List<User>(users);

```

Comment:
another option to do this would be to do `users.ToList()` i think this should work 🤔